111-use-buildah-instead-of-kaniko.md

```diff
@@ -0,0 +1,236 @@
+# Buildah - replacement of Kaniko after its archivation
+
+In our `KafkaConnect` custom resource, we have a possibility to specify plugins that we want to include inside our Connect and build the Connect image, without need of building the image externally.
+To support this feature on both Kubernetes and OpenShift, we have two implementations.
+For OpenShift we are using the Build API that OpenShift offers.
+But on Kubernetes, we are using [Kaniko](https://github.com/GoogleContainerTools/kaniko).
+However, the Kaniko project was [archived](https://github.com/GoogleContainerTools/kaniko/commit/236ba5690eda9170d0157aa8137ebbeb09d38685) and it's no longer developed or maintained.
+This proposal aims to resolve this issue.
+
+## Motivation
+
+The main motivation behind this proposal is to resolve the issue that Kaniko is archived, and it will not receive any updates (no bug or CVE fixes).
+Currently, it seems to not be a problem, but when some critical CVE or bug arises, we have to fix it.
+Also, we need a support in case that there will be something blocking us on newer versions of Kubernetes, which we would not get in this case anymore.
+
+## Proposal
+
+For the Connect Build feature on Kubernetes, we will use [Buildah](https://buildah.io/) as a replacement of Kaniko.
+Buildah is well-supported, widely used tool for building container images.
+It also appears that OpenShift’s Build API relies on Buildah, which makes Buildah a strong alternative to Kaniko, with the added assurance of long-term support and maintenance.
+Other than that, Buildah doesn't need any workaround in order to run it on Kubernetes rootlessly - the only thing that is needed during the build and push stages is to specify `--storage-driver=vfs` option.
+VFS (Virtual File System) driver ensures user-space implementation, it stores each layer as a full copy of the files, but without need of root permissions.
+The only trade-off is the build time (it's slower) and consumption of disk space.
+
+Even though we will run Buildah with these options, which should make it rootless, it still needs some capabilities to run without issues during the build phase.
+That means we cannot run the build Pod with restricted Pod Security profile, as these capabilities are dropped in these profiles.
+This is more described in the [Running Buildah in restricted environment](#running-buildah-in-restricted-environment) section of this proposal.
+
+We will use the official Buildah images on Quay.io - currently `quay.io/buildah/stable:v1.40.1` - and similarly to Kaniko executor image - we will pull the official image, tag it with the version and push it to the Strimzi repository on Quay.
+That's useful because of our versioning and we will have the image ready even if someone on Buildah side decides to remove it from their Quay repository.
```

Comment:
```suggestion
That's useful because of our versioning. We will have the image ready even if someone on the Buildah side decides to remove it from their Quay repository.
```